ang/crmMosaico/TemplateItem.js

```diff
@@ -12,13 +13,26 @@
       },
       link: function($scope, $el, $attr) {
         var ts = $scope.ts = CRM.ts('mosaico');
-        $scope.$watch('crmMosaicoTemplateItem', function(newValue){
+
+        $scope.$watch('crmMosaicoTemplateItem', function(newValue) {
           $scope.myOptions = newValue;
+
+          // Template default action label based on current state
+          if (newValue['state']) {
+            var clickActionLabels = {
+              select: "Select",
+              selected: "Edit",
+              configure: "Edit",
+              new: "New"
+            };
+            var clickActionLabel = clickActionLabels[newValue['state']];
+            $scope.actionLabel = clickActionLabel;
```

Comment:
isn't there a more meaningful name to give to this variable? You have multiple actions on the directive, it just so happens that only one of those have a variable name. Can you identify it with a more specific name?